test/unittest/blas1/blas1_iaminmax_common.hpp

```diff
@@ -0,0 +1,79 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename blas1_iaminmax_common.hpp
+ *
+ **************************************************************************/
+
+#ifndef BLAS_TEST_IAMINMAX_COMMON_HPP
+#define BLAS_TEST_IAMINMAX_COMMON_HPP
+
+#include "blas_test.hpp"
+
+enum class generation_mode_t : char {
+  Random = 'r',
+  Limit = 'l',
+  Incrementing = 'i',
+  Decrementing = 'd'
+};
+
+using combination_t = std::tuple<int, int, generation_mode_t>;
+
+template <typename scalar_t>
+void populate_data(generation_mode_t mode, scalar_t limit,
+                   std::vector<scalar_t> &vec) {
+  switch (mode) {
+    case generation_mode_t::Random:
+      fill_random(vec);
+      break;
+    case generation_mode_t::Limit:
+      std::fill(vec.begin(), vec.end(), limit);
```

Comment:
What will the value of fill will be here?

Reply:
Left a comment on `generation_mode_t::Limit`.

Reply:
`limit`, which is passed by the iamin or iamax test